rfd/0073-discover.md

```diff
@@ -0,0 +1,337 @@
+  we currently support auto-discovery for: EC2, RDS/Aurora, Redshift, Elasticache,
+  MemoryDB. The dialog may optionally allow to specify tags to filter the
+  resources by.
+- Based on the selected resources, display appropriate `teleport discover aws`
+  command to run and the IAM policy required for the command to successfully
+  discover and join resources.
+
+Users will launch the `teleport discover aws` command on an AWS instance that
+has IAM permissions required to discover and connect resoures. We'll detail
+those permissions later.
+
+The command will:
+
+1. Join the Teleport cluster as a "discover" agent using the same cluster join
+   mechanism as any other agent. Join token will be generated by the server
+   and included in the CLI command displayed to a user.
+2. Using the node's IAM credentials, search the cloud account for the requested
+   resources according to the command's filters.
+3. Once discovered, the agent will start heartbeating the discovered resources
+   to the cluster with TTL using the same heartbeat mechanism as any other
+   type of resource. Heartbeating resources to the cluster will allow the UI
+   to pick them up and show to the user.
+4. Wait for the user to inspect discovered resources in the UI wizard and select
+   those they would like to connect.
+5. Once confirmed, proceed with importing resources to the cluster. Details
+   below on how this works for each particular resource type.
+6. After importing all resources, `teleport discover` will generate Teleport
+   config file `/etc/teleport.yaml`, start the Teleport service on the node
+   and exit.
+
+## Details
+
+### 1. Join
+
+Running as an agent and joining the Teleport cluster allows `teleport discover`
+to share progress with the UI wizard served by the Proxy. The command will also
+have a CLI-only mode which will be detailed later.
+
+Discover agent will use the same cluster join mechanism as any other type of
+agent e.g. SSH node, application or database service. A new auth token type
+`discover` will be introduced for the discover agent. Users will be able to
+create it same way as any other token type:
+
+```
+$ tctl tokens add --type=discover
+```
+
+The `teleport discover` command shown to a user by the UI wizard will include
+a pre-generated short-lived auth token:
+
+```
+$ teleport discover aws --proxy=proxy.example.com:443 --token=abc123
+```
+
+When running `teleport discover` manually in a CLI mode, users will generate
+dynamic auth token using `tctl tokens add` command.
+
+### 2. Search
+
+The `teleport discover` command will use default credential provider chain to
+determine the cloud account credentials. For AWS this includes IAM role, then
+environment variables, then shared credentials file - all of which is handled
+by the SDK automatically.
+
+`teleport discover` will use respective `Describe*` AWS SDK methods to scan
+the cloud account for the requested resources in the requested regions.
+
+```
+$ teleport discover aws \
```

Comment:
Can this command fail gracefully, for example

[1] Checking your AWS permissions
[2] This user is missing permissions, .... apply this policy for user to proceed